Add support for optional parameters in a test method

hughbe opened the pull request:
I seem to remember seeing an issue in the repo requesting something like this but can't find it now!

Also: http://stackoverflow.com/questions/14818439/xunit-test-engines-inlinedataattribute-optional-method-parameters

/cc @bradwilson

dnfclas commented:
Hi __@hughbe__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

hughbe commented:
Hmm, how odd that the tests are failing: using TestDriven.NET to run tests (as per the recommended contributor guidelines), I get all passes
![tests](https://cloud.githubusercontent.com/assets/1275900/19362799/eb5e71be-917f-11e6-8efa-e15ee0de08a7.png)

@bradwilson is there some sort of weird CI stuff going on? Could you take a look when you have time :)

hughbe commented:
Okay, so it turns out xunit has two different runners of theory tests! I had only managed to test one of them from TestDriven.net, but the CI tests a different one.

I've fixed this and removed some code that isn't necessary at all

hughbe commented:
@bradwilson can you take a look if you have time - I keep coming up with places I wanna use this in my tests, thanks!

bradwilson commented on test/test.xunit.execution/Acceptance/Xunit2TheoryAcceptanceTests.cs:
```diff
@@ -14,6 +13,102 @@ public class Xunit2TheoryAcceptanceTests
+            }
+
+            [Theory]
+            [InlineData()]
```
Never thought I'd see this. :-p

bradwilson commented:
Merged, thanks!
